src/back/msl/writer.rs

```diff
@@ -829,9 +842,23 @@ impl<W: Write> Writer<W> {
                     if !is_scoped {
                         write!(self.out, "(")?;
                     }
-                    self.put_expression(left, context, false)?;
+                    // Because packed vectors such as `packed_float3` cannot be directly multipied by
+                    // matrices, we them in the non-packed variant first.
+                    if is_tri_vec_mul_matrix {
```

Comment:
could use a helper function `self.put_mul_expression` or something, to ease on the stack size